nexustreetojson.py

```diff
@@ -12,30 +25,59 @@ def _is_array(possible_array):
 
 
 def _attrs_to_json(tree):
-    names = sorted(tree.attrs)
+    """
+    Creates a list of lines of JSON for the attributes of a given group or dataset
+
+    :param tree: Group or dataset tree node
+    :return: List of lines of JSON
+    """
     result = []
-    for k in names:
-        if nexusformat.nexus.tree.is_text(tree.attrs[k]):
-            attr_string = u'": "' + nexusformat.nexus.tree.text(tree.attrs[k]) + '"'
-        elif _is_array(tree.attrs[k]):
-            array_string = np.array2string(tree.attrs[k], separator=', ').replace(' ', '')
-            array_string = array_string.replace('.,', '.0,')
-            array_string = array_string.replace('.]', '.0]')
-            attr_string = u'": ' + array_string
-        else:
-            attr_string = u'": ' + nexusformat.nexus.tree.text(tree.attrs[k])
-        txt = (u'{"' + k + attr_string).replace("u'", "'")
-        txt += '}'
-        result.append(txt)
-    result[0] = '\"attributes\": [' + result[0]
-    result[-1] += ']'
-    return ', '.join(result)
+    if tree.attrs:
+        for k in tree.attrs:
+            if nexusformat.nexus.tree.is_text(tree.attrs[k]):
+                attr_string = u'": "' + nexusformat.nexus.tree.text(tree.attrs[k]) + '"'
+            elif _is_array(tree.attrs[k]):
+                array_string = np.array2string(tree.attrs[k], separator=', ').replace(' ', '')
+                array_string = array_string.replace('.,', '.0,')
+                array_string = array_string.replace('.]', '.0]')
```

Comment:
I wondered whether there is a cleaner way to do this, but I guess you've searched already for that. I looked at <https://docs.scipy.org/doc/numpy/reference/generated/numpy.set_printoptions.html#numpy.set_printoptions> but couldn't find something at first sight.

Reply:
There might be, but I'm not too concerned; this is a quick prototype. I'm mostly interested in whether the format it outputs is correct.
I talked with Nikhil yesterday and he now has something similar in NICOS. I think it is worthwhile having something independent of NICOS; I could then use it with my NeXus reader to do a test from file through Kafka and back into file for example. Nikhil's version already has functionality to insert stream commands into the JSON, so I might take his and update it to Python 3.

Thanks for taking a look at this. The `NXclass` thing you spotted it actually a mistake in the NeXus file itself.